src/Mapper/DoctrineCollector.php

```diff
@@ -95,11 +95,11 @@ public function addDiscriminator(string $class, string $key, string $discriminat
     public function addDiscriminatorColumn(string $class, ColumnDefinitionBuilder $columnDef): void
     {
         if (!isset($this->discriminatorColumns[$class])) {
-            $this->discriminatorColumns[$class] = $columnDef;
+            $this->discriminatorColumns[$class] = $columnDef->getOptions();
         }
     }
 
-    public function addInheritanceType(string $class, string $type): void
+    public function addInheritanceType(string $class, int $type): void
```

Comment:
Had to add another BC breack, this type should be int, otherwise it does not match with the DoctrineORMMapper (https://github.com/sonata-project/sonata-doctrine-extensions/blob/1.x/src/Mapper/ORM/DoctrineORMMapper.php#L105) or the   ClassMetadataInfo (https://github.com/doctrine/orm/blob/v2.7.2/lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php#L69)

Reply:
Can you find a BC way for this?